amethyst_core/src/transform/components/local_transform.rs

```diff
@@ -33,6 +42,122 @@ impl LocalTransform {
+    /// by whatever amount you want to move before passing the vector to this method
+    #[inline]
+    pub fn move_global(&mut self, direction: Vector3<f32>) -> &mut Self {
+        self.translation = (Vector3::from(self.translation) + direction).into();
+        self
+    }
+
+    /// Move relatively to its current position and orientation.
+    #[inline]
+    pub fn move_local(&mut self, axis: Vector3<f32>, amount: f32) -> &mut Self {
+        let delta = Quaternion::from(self.rotation).conjugate() * axis.normalize() * amount;
+
+        self.translation = (Vector3::from(self.translation) + delta).into();
+        self
+    }
+
+    /// Move relatively to its current position and orientation.
+    pub fn move_right(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
+        self.move_local(orientation.right.into(), amount)
+    }
+
+    /// Move relatively to its current position and orientation.
+    pub fn move_up(&mut self, orientation: &Orientation, amount: f32) -> &mut Self {
+        self.move_local(orientation.up.into(), amount)
+    }
+
+    /// Pitch relatively to the world.
+    pub fn pitch_global(&mut self, orientation: &Orientation, angle: Deg<f32>) -> &mut Self {
+        self.rotate_global(orientation.right.into(), angle)
+    }
+
+    /// Pitch relatively to its own rotation.
+    pub fn pitch_local(&mut self, orientation: &Orientation, angle: Deg<f32>) -> &mut Self {
+        self.rotate_local(orientation.right.into(), angle)
+    }
+
+    /// Roll relatively to the world.
+    pub fn roll_global(&mut self, orientation: &Orientation, angle: Deg<f32>) -> &mut Self {
+        self.rotate_global(orientation.forward.into(), angle)
+    }
+
+    /// Roll relatively to its own rotation.
+    pub fn roll_local(&mut self, orientation: &Orientation, angle: Deg<f32>) -> &mut Self {
+        self.rotate_local(orientation.forward.into(), angle)
+    }
+
+    /// Add a rotation to the current rotation
+    #[inline]
+    pub fn rotate(&mut self, quat: Quaternion<f32>) -> &mut Self {
+        self.rotation = (quat * Quaternion::from(self.rotation)).into();
+        self
+    }
+
+    /// Rotate relatively to the world
+    #[inline]
+    pub fn rotate_global(&mut self, axis: Vector3<f32>, angle: Deg<f32>) -> &mut Self {
+        let axis_normalized = Vector3::from(axis).normalize();
+        let q = Quaternion::from_axis_angle(axis_normalized, angle);
+
+        self.rotate(q)
+    }
+
+    /// Rotate relatively to the current orientation
+    #[inline]
+    pub fn rotate_local(&mut self, axis: Vector3<f32>, angle: Deg<f32>) -> &mut Self {
+        let rel_axis_normalized = Quaternion::from(self.rotation)
+            .rotate_vector(Vector3::from(axis))
+            .normalize();
+        let q = Quaternion::from_axis_angle(rel_axis_normalized, angle);
+
+        self.rotate(q)
+    }
+
+    /// Set the position.
+    pub fn set_position(&mut self, position: Point3<f32>) -> &mut Self {
```

Comment:
E.g. here I have the feeling that exposing the cgmath type just makes it more verbose.